x-pack/solutions/search/plugins/search_playground/public/components/query_mode/query_side_panel.tsx

```diff
@@ -0,0 +1,188 @@
+import { ChatForm, ChatFormFields } from '../../types';
+import { AnalyticsEvents } from '../../analytics/constants';
+import { docLinks } from '../../../common/doc_links';
+import { createQuery } from '../../utils/create_query';
+import { SearchQuery } from './search_query';
+
+const isQueryFieldSelected = (
+  queryFields: ChatForm[ChatFormFields.queryFields],
+  index: string,
+  field: string
+): boolean => {
+  return Boolean(queryFields[index]?.includes(field));
+};
+
+export const QuerySidePanel = () => {
+  const usageTracker = useUsageTracker();
+  const { fields } = useSourceIndicesFields();
+  const sourceFields = useWatch<ChatForm, ChatFormFields.sourceFields>({
+    name: ChatFormFields.sourceFields,
+  });
+  const {
+    field: { onChange: queryFieldsOnChange, value: queryFields },
+  } = useController<ChatForm, ChatFormFields.queryFields>({
+    name: ChatFormFields.queryFields,
+  });
+  const {
+    field: { onChange: elasticsearchQueryChange },
+  } = useController<ChatForm, ChatFormFields.elasticsearchQuery>({
+    name: ChatFormFields.elasticsearchQuery,
+  });
+
+  const updateFields = (index: string, fieldName: string, checked: boolean) => {
+    const currentIndexFields = checked
+      ? [...queryFields[index], fieldName]
+      : queryFields[index].filter((field) => fieldName !== field);
+    const updatedQueryFields = { ...queryFields, [index]: currentIndexFields };
+
+    queryFieldsOnChange(updatedQueryFields);
+    elasticsearchQueryChange(createQuery(updatedQueryFields, sourceFields, fields));
+    usageTracker?.count(AnalyticsEvents.queryFieldsUpdated, currentIndexFields.length);
+  };
+
+  return (
+    <EuiPanel color="subdued" hasShadow={false}>
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiText>
+          <h5>
+            <FormattedMessage
+              id="xpack.searchPlayground.viewQuery.sideBar.query.title"
+              defaultMessage="Query"
+            />
+          </h5>
+        </EuiText>
+        <EuiPanel grow={false} hasShadow={false} hasBorder>
+          <SearchQuery />
+        </EuiPanel>
+        <EuiText>
+          <h5>
+            <FormattedMessage
+              id="xpack.searchPlayground.viewQuery.flyout.table.title"
+              defaultMessage="Fields to search (per index)"
+            />
+          </h5>
+        </EuiText>
+        {Object.entries(fields).map(([index, group], indexNum) => (
+          <EuiFlexItem grow={false} key={index}>
+            <EuiPanel grow={false} hasShadow={false} hasBorder>
+              <EuiAccordion
+                id={index}
+                buttonContent={
+                  <EuiText>
+                    <h5>{index}</h5>
+                  </EuiText>
+                }
+                initialIsOpen
+                data-test-subj={`fieldsAccordion-${indexNum}`}
+              >
+                <EuiSpacer size="s" />
+
+                <EuiBasicTable
+                  tableCaption="Query Model table"
```

Comment:
missing i18n.

Reply:
Good catch, I didn't even check for that as I was just abstracting what was already present.